pkg/ansible/proxy/requestfactory/requestinfo.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This code was retrieved from
+// https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/request/requestinfo.go
+// and slightly modified for use in this project
+
+package requestfactory
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/validation/path"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"k8s.io/klog"
```

Comment:
we should change this to use logrus

Reply:
I used `logf` from controller-runtime since that is what the proxy is using, is there anything wrong with that approach? Wasn't sure the benefits of using logr.